mail/views_test.py

```diff
@@ -140,43 +146,106 @@ def test_automatic_email(self):
+        assert mock_get_emails.call_args[0][0].to_dict() == search_obj.to_dict()
 
         assert mock_mailgun_client.send_batch.called
         _, called_kwargs = mock_mailgun_client.send_batch.call_args
         assert called_kwargs['subject'] == self.request_data['email_subject']
         assert called_kwargs['body'] == self.request_data['email_body']
         assert called_kwargs['recipients'] == email_results
 
-        assert AutomaticEmail.objects.count() == 1
-        automatic_email = AutomaticEmail.objects.first()
-        percolate_query = automatic_email.query
-        mocked_index_percolate_queries.assert_called_with([percolate_query.id])
-        assert percolate_query.query == search_dict
-        assert automatic_email.email_subject == self.request_data['email_subject']
-        assert automatic_email.email_body == self.request_data['email_body']
-        assert automatic_email.sender_name == full_name(self.staff)
-        assert automatic_email.enabled is True
+        assert mock_add_automatic_email.call_args[0][0].to_dict() == search_obj.to_dict()
+        assert mock_add_automatic_email.call_args[1] == {
+            "email_subject": self.request_data['email_subject'],
+            "email_body": self.request_data['email_body'],
+            "sender_name": full_name(self.staff),
+        }
+
+        assert SentAutomaticEmail.objects.filter(
+            user__email__in=email_results, automatic_email=automatic_email,
+        ).count() == len(email_results)
+
+    def test_automatic_email_fail(self):
+        """
+        If one request fails, we should mark all other success emails as having been sent anyway
+        """
+        email_results = set(['a@example.com', 'b@example.com'])
+        request_data = self.request_data.copy()
+        request_data['send_automatic_emails'] = True
+        for email in email_results:
+            UserFactory.create(email=email)
+
+        automatic_email = AutomaticEmail.objects.create(
+            email_subject='subject',
+            email_body='body',
+            query=None,
+            sender_name='sender',
+        )
+        search_obj = create_search_obj(
+            user=self.staff,
+            search_param_dict=request_data['search_request'],
+            filter_on_email_optin=True,
+        )
+
+        success_emails = {'a@example.com'}
+        failure_emails = email_results.difference(success_emails)
+        exception_pairs = [
+            (failure_emails, HTTPError())
+        ]
+
+        with patch(
+            'mail.views.get_all_query_matching_emails', autospec=True, return_value=email_results
+        ), patch(
+            'mail.views.MailgunClient'
+        ) as mock_mailgun_client, patch(
+            'mail.views.add_automatic_email', autospec=True, return_value=automatic_email,
+        ) as mock_add_automatic_email:
+            mock_mailgun_client.send_batch.side_effect = SendBatchException(exception_pairs)
```

Comment:
You're not able to set this side_effect in the actual `patch()` call? Also, what exactly are we testing for with the `assert send_batch_exception` line below?